tests/testthat/testCombine.R

```diff
@@ -0,0 +1,101 @@
+# testCombine.R
+
+context("test for Combine SNV and CNA files")
+
+# ==== BEGIN SETUP AND PREPARE =================================================
+# set up tempdir and tempfiles for output files and to be deleted after tests
+OLOG <- as.character(getOption("rete.logfile"))   # save original logfile name
+logFileName(fPath = tempdir(), setOption = TRUE)  # make tempdir() the log dir
+logName <- unlist(getOption("rete.logfile"))
+if (file.exists(logName)) { file.remove(logName)}
+# ==== END SETUP AND PREPARE ===================================================
+
+test_that("parameter errors are correctly handled", {
+    # try no parameter input
+    expect_error(combineSNV_CNA(writeLog=FALSE),
+                 'Input vectors are empty')
+
+    # try wrong file format (e.g. text file instead of rds dataframe) in SNV input vector
+    input_vector = c("SNV.txt", "CNA.rds")
+    expect_error(combineSNV_CNA(input_vector[1], input_vector[2], writeLog=FALSE),
+                 'SNV file has to be .rds format.')
+
+    # try wrong file format (e.g. text file instead of rds dataframe) in CNA input vector
+    input_vector = c("SNV.rds", "CNA.txt")
+    expect_error(combineSNV_CNA(input_vector[1], input_vector[2], writeLog=FALSE),
+                 'CNA file has to be .rds format.')
+
+    # sane input vector
+    input_vector = c("SNV.rds", "CNA.rds")
+    # try non existing input file for SNV input vector
+    expect_error(combineSNV_CNA(c("nonExistent.rds"), input_vector[2], writeLog=FALSE),
+                 '.checkArgs> "fSNV" error: file "nonExistent.rds" does not exist')
+
+    # try non existing input file for CNA input vector
+    expect_error(combineSNV_CNA(input_vector[1], c("nonExistent.rds"), writeLog=FALSE),
+                 '.checkArgs> "fCNA" error: file "nonExistent.rds" does not exist')
+
+    # try invalid output fgX file name (e.g. wrong extension)
+    expect_error(combineSNV_CNA(input_vector[1], input_vector[2], "out.txt", writeLog=FALSE),
+                 'Output file has to be .rds extension!')
+
+    # try null output fgX file name
+    expect_error(combineSNV_CNA(input_vector[1], input_vector[2], NULL, writeLog=FALSE),
+                 'argument is of length zero')
+
+    # try null silent option
+    expect_error(combineSNV_CNA(input_vector[1], input_vector[2], silent=NULL, writeLog=FALSE),
+                 '.checkArgs> "silent" mode error')
+
+    # try null writeLog option
+    expect_error(combineSNV_CNA(input_vector[1], input_vector[2], writeLog=NULL),
+                 '.checkArgs> "writeLog" mode error')
+
+    # check if log file has been written
+    expect_false(file.exists(logName))
+})
+
+
+test_that("a sane input gives an expected output", {
+    # provide one valid SNV and one valid CNA file with small number of genes as input
+    # TODO where can I find the sample data for testing? inst/extdata folder has maf & txt formats
```

Comment:
Post  a request on the mailing list - even though the various functions are still being developed, people should be far enough along that they could just create a sample dataset with the  [minimal dataset](http://steipe.biochemistry.utoronto.ca/abc/students/index.php/User:Naina.Singh/Validation_Dataset).